client/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import Joi from 'joi';
+
+import {
+  Avatar,
+  Button,
+  CssBaseline,
+  TextField,
+  Link,
+  Grid,
+  Box,
+  Typography,
+  Container,
+} from '@mui/material';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import axios from 'axios';
+import BasicSelect from '../components';
+
+const Copyright = (props: any) => {
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {'Copyright © '}
+      <Link color="inherit" href="/">
+        Menu Mingle
+      </Link>
+      {new Date().getFullYear()}.
+    </Typography>
+  );
+};
+
+const serverUrl = import.meta.env.VITE_APP_SERVER_URL;
+
+const SignUp = () => {
+  const [role, setRole] = React.useState();
+  const [errors, setErrors] = React.useState({});
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+
+    const schema = Joi.object({
+      username: Joi.string().min(3).max(30).required().lowercase(),
+      password: Joi.string()
+        .min(8)
+        .max(30)
+        .required()
+        .pattern(
+          /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
+        ),
+      role: Joi.number().required(),
+    });
+
+    const formData = {
+      username: data.get('username'),
+      password: data.get('password'),
+      role: +role,
+    };
+
+    const result = schema.validate(formData);
+
+    const { error } = result;
+    if (error) {
+      // Handle validation error
+      setErrors(error.details[0]);
+    } else {
+      // Data is valid, proceed with submission
+      try {
+        const res = await axios.post(`${serverUrl}/api/v1/auth/register`, {
+          username: data.get('username'),
+          password: data.get('password'),
+          roleId: +role,
+        });
+
+        window.location.href = '/login';
+      } catch (err: any) {
+        throw new Error(err);
+      }
+    }
+  };
+
+  return (
+    <>
```

Comment:
Add a div, section, or anything here and attach a background color from our theme to it.

Reply:
@moustf  after a long trouble with css 😒